src/text_agent.rs

```diff
@@ -0,0 +1,245 @@
+//! The text agent is an `<input>` element used to trigger
+//! mobile keyboard and IME input.
+
+use std::{cell::Cell, rc::Rc, sync::Mutex};
+
+use bevy::{
+    prelude::{EventWriter, Res, Resource},
+    window::RequestRedraw,
+};
+use crossbeam_channel::Sender;
+
+use once_cell::sync::Lazy;
+use wasm_bindgen::prelude::*;
+
+use crate::systems::ContextSystemParams;
+
+static AGENT_ID: &str = "egui_text_agent";
+
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug, Default)]
+pub struct VirtualTouchInfo {
+    pub editing_text: bool,
+}
+
+pub static VIRTUAL_KEYBOARD_GLOBAL: Lazy<Mutex<VirtualTouchInfo>> =
+    Lazy::new(|| Mutex::new(VirtualTouchInfo::default()));
```

Comment:
Can we use stdlib's [`OnceCell`](https://doc.rust-lang.org/std/cell/struct.OnceCell.html) instead?

Reply:
The value gets set multiple times so I don't think so.

Reply:
It can contain `Mutex` the same way `Lazy` does rn.

Reply:
I think the std `OnceCell` doesn't let you implement `Sync` and the std `OnceLock` only lets you do `static` so this is actually necessary despite the lack of true threads.

Reply:
I removed it and added a minimum rust version to use std sync `LazyLock` I think I need a 'static....